src/main/java/com/dev/movietheatre/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.dev.movietheatre.dao.impl;
+
+import com.dev.movietheatre.dao.MovieDao;
+import com.dev.movietheatre.lib.Dao;
+import com.dev.movietheatre.model.Movie;
+import com.dev.movietheatre.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert Movie entity", e);
```

Comment:
```suggestion
            throw new RuntimeException("Can't insert Movie entity" + movie, e);
```
and add own exception

Reply:
done